File: family_finance_api/FamilyFinance/Models/Entities/Budget.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace FamilyFinance.Models.Entities;

[Table("Budgets")]
public class Budget : BaseModel<int>
{
    [Column("category_id")]
    public int CategoryId { get; set; }

    [Column("start_date")]
    public DateTime StartDate { get; set; }

    [Column("end_date")]
    public DateTime EndDate { get; set; }

    [Column("amount")]
    public decimal Amount { get; set; }

    [ForeignKey(nameof(CategoryId))]
    public virtual Category Category { get; set; } = null!;
}

File: family_finance_api_go/internal/models/budget.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Budget struct {
	StartDate time.Time `gorm:"column:start_date;type:date" json:"start_date"`
	EndDate   time.Time `gorm:"column:end_date;type:date" json:"end_date"`
	Amount    float64   `gorm:"column:amount;type:decimal(18,2)" json:"amount"`

	CategoryID uint `gorm:"column:category_id"`

	Category *Category `gorm:"foreignKey:CategoryID;references:ID"`

	gorm.Model
}
